site/_data/deprecationCalendar.js

```diff
@@ -0,0 +1,73 @@
+const path = require('path');
+const fs = require('fs').promises;
+const yaml = require('js-yaml');
+
+/**
+ * Pulls a list of deprecataion items
+ *
+ * @return {promise} A promise that resolves in an array containing all the deprecations
+ */
+async function getDeprecationData() {
+  let deprecations;
+  const deprecationsFile = path.join(
+    __dirname,
+    '../../external/data/deprecation-calendar.json'
+  );
+
+  try {
+    deprecations = JSON.parse(await fs.readFile(deprecationsFile, 'utf-8'));
+  } catch (error) {
+    console.error('Error reading Deprecations file', error);
+  }
+
+  return deprecations;
+}
+
+/**
+ * Pulls articles related to deprecations from the deprecation-articles.yml file
+ *
+ * @return {promise} A promise that resolves in an array containing all the articles
+ */
+async function getArticles() {
+  let articles;
+  const articlesFile = path.join(
```

Comment:
Misleading variable name, that's only the path, as this will always be the same consider moving it to the top of the file like `const RELATED_ARTICLES_PATH = ...`.